small_string/lib.rs

```diff
@@ -11,16 +13,16 @@ impl std::fmt::Debug for SmallString {
 
 impl SmallString {
     pub const EMPTY: SmallString = Self {
-        bytes: [0, 0, 0, 0, 0, 0, 0],
+        bytes: [0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF],
     };
 
     pub fn len(&self) -> usize {
-        // Find the last non-null character and add one to its index to get length.
+        // Find the last non-0xFF character and add one to its index to get length.
         self.bytes
             .as_slice()
             .iter()
             .rev()
-            .position(|&x| x != 0)
+            .position(|&x| x != 0xFF)
```

Comment:
**nitpick:** We could now turn this logic back around to look for the first 0xFF. Would be a bit easier to understand.